client/wrappers/metered/matching_generated.go

```diff
@@ -1,280 +1,191 @@
+
 // Permission is hereby granted, free of charge, to any person obtaining a copy
 // of this software and associated documentation files (the "Software"), to deal
 // in the Software without restriction, including without limitation the rights
 // to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 // copies of the Software, and to permit persons to whom the Software is
 // furnished to do so, subject to the following conditions:
 //
-// The above copyright notice and this permission notice shall be included in
-// all copies or substantial portions of the Software.
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
 //
 // THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 // IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 // FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 // AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-// THE SOFTWARE.
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
 
-package matching
+package metered
+
+// Code generated by gowrap. DO NOT EDIT.
+// template: ../../templates/metered.tmpl
+// gowrap: http://github.com/hexdigest/gowrap
 
 import (
 	"context"
-	"strings"
 
 	"go.uber.org/yarpc"
 
-	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/client/matching"
 	"github.com/uber/cadence/common/metrics"
 	"github.com/uber/cadence/common/types"
 )
 
-var _ Client = (*metricClient)(nil)
-
-type metricClient struct {
-	client        Client
+// matchingClient implements matching.Client interface instrumented with retries
+type matchingClient struct {
+	client        matching.Client
 	metricsClient metrics.Client
 }
 
-// NewMetricClient creates a new instance of Client that emits metrics
-func NewMetricClient(client Client, metricsClient metrics.Client) Client {
-	return &metricClient{
+// NewMatchingClient creates a new instance of matchingClient with retry policy
+func NewMatchingClient(client matching.Client, metricsClient metrics.Client) matching.Client {
+	return &matchingClient{
 		client:        client,
 		metricsClient: metricsClient,
 	}
 }
 
-func (c *metricClient) AddActivityTask(
-	ctx context.Context,
-	request *types.AddActivityTaskRequest,
-	opts ...yarpc.CallOption,
-) error {
-	c.metricsClient.IncCounter(metrics.MatchingClientAddActivityTaskScope, metrics.CadenceClientRequests)
-	sw := c.metricsClient.StartTimer(metrics.MatchingClientAddActivityTaskScope, metrics.CadenceClientLatency)
+func (c *matchingClient) AddActivityTask(ctx context.Context, ap1 *types.AddActivityTaskRequest, p1 ...yarpc.CallOption) (err error) {
 
-	c.emitForwardedFromStats(
```

Comment:
from a bit of idle thinking... maybe something like this would work?
```go
func (c *client) emitForwardedFromStats(scope int, request any) {
  r, ok := request.(interface{GetForwardedFrom() string})
  if !ok { return }
  // do stuff
}
```
and then just call it in all methods.

kinda seems like we may want this on _any_ forwardable request, and if we added future ones we'd want it there too.

Reply:
good catch. I'll restore it.

Reply:
Updated